mitmproxy/addons/next_layer.py

```diff
@@ -239,6 +247,12 @@ def _ignore_connection(
         else:  # pragma: no cover
             raise AssertionError()
 
+    @staticmethod
+    def _extract_host(data_client: bytes) -> str:
```

Comment:
```suggestion
    def _extract_http1_host_header(data_client: bytes) -> str:
```
(I'd like to emphasize this is for HTTP and HTTP/1 only)